packages/flutter/lib/src/widgets/draggable_scrollable_sheet.dart

```diff
@@ -298,12 +353,38 @@ class _DraggableScrollableSheetState extends State<DraggableScrollableSheet> {
     _extent = _DraggableSheetExtent(
```

Comment:
@dnfield It's just occurred to me that this code in prod and in my branch is a bit broken:
`DraggableScrollableSheet` does not update the extent when the widget is rebuilt with new values. This means that the extent will retain potentially invalid values for min, max, snapSizes, etc. if the user rebuilds the sheet with new parameters.

We need to override `DidUpdateWidget` to rebuild the extent when necessary. This feels fairly non-trivial and is largely about fixing existing behavior in prod-are you okay with me leaving this fix to a follow up PR to go in after this one merges? Or should I spin up a new PR now to fix it and rebase this one on top of it? Seems like a fairly severe bug that may justify the latter.

Reply:
Here's a test that can be added to `draggable_scrollable_sheet_test.dart` to demonstrate the issue:
```dart
    testWidgets('A changed snapSizes list is propagated', (WidgetTester tester) async {
      const Key stackKey = ValueKey<String>('stack');
      const Key containerKey = ValueKey<String>('container');
      await tester.pumpWidget(_boilerplate(null,
        snap: true,
        stackKey: stackKey,
        containerKey: containerKey,
        snapSizes: <double>[.5],
      ));
      await tester.pumpAndSettle();
      final double screenHeight = tester.getSize(find.byKey(stackKey)).height;

      await tester.drag(find.text('Item 1'), Offset(0, -.4 * screenHeight));
      await tester.pumpAndSettle();
      expect(
        tester.getSize(find.byKey(containerKey)).height / screenHeight,
        closeTo(1.0, precisionErrorTolerance),
      );

      // Pump the same widget but with a new snapSizes, triggering a rebuild.
      await tester.pumpWidget(_boilerplate(null,
        snap: true,
        stackKey: stackKey,
        containerKey: containerKey,
        maxChildSize: .8,
        snapSizes: <double>[.5],
      ));
      await tester.pumpAndSettle();

      // The max child size has been reduced, we should be rebuilt at the new
      // max of .8.
      expect(
        tester.getSize(find.byKey(containerKey)).height / screenHeight,
        closeTo(.8, precisionErrorTolerance),
      );
    }, variant: TargetPlatformVariant.all());
```

Reply:
Doing this in a new PR is fine. https://github.com/flutter/flutter/pull/69724/files probbly would fix this, and relanding that without the bit about caching the child is probably a good idea.